### ISSUE TITLE
Sample code fails shows TS error: Property 'fibonacci' in type 'MathService' is not assignable to the same property in base type

### ISSUE THREAD

**Issue by CharlieDigital** [StyleT/webext-pegasus#4]
Copied the sample code:

```ts
import {IPegasusRPCService, PegasusRPCMessage} from '@webext-pegasus/rpc';

export type IMathService = InstanceType<
  typeof MathService
>;

export class MathService
  implements IPegasusRPCService<MathService>
{
  fibonacci(num: number): number {
    return (num <= 1) ? 1 : this.fibonacci(num - 1) + this.fibonacci(num - 2);
  }
}
```
Get the following error:

<img width="533" alt="image" src="https://github.com/user-attachments/assets/72c33079-3c1b-4a99-9f05-802f749defb1">

```
Property 'fibonacci' in type 'MathService' is not assignable to the same property in base type 'IPegasusRPCServiceInternal<MathService>'.
  Type '(num: number) => number' is not assignable to type 'GoodFunc<(num: number) => number>'.
    Types of parameters 'num' and 'args' are incompatible.
      Type 'unknown' is not assignable to type 'number'
```

VS Code reporting TypeScript version 5.5.2

**Comment by StyleT** [comment]
Sorry for late reply, will take a good look tomorrow

**Comment by CharlieDigital** [comment]
Nice!

**Comment by StyleT** [comment]
Published as `v0.3.6`. Thanks for your contribution!
